### PR TITLE
Put the `grant_access_to` limitation behind a version block to avoid confusion

### PR DESCRIPTION
**Pull request by mikealfare**
## What are you changing in this pull request and why?

`dbt-bigquery` has a limitation that affects only versions prior to 1.4. We communicated this by indicating what the issue was for old versions regardless of which version was selected in the version drop down. This created confusion for folks who were looking at a newer version of the docs and still saw the limitation.

## Checklist
- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [x] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [x] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

### CODE REVIEW

**Comment by mirnawong1** [comment]
slack thread: https://dbt-labs.slack.com/archives/C067EUS2D7S/p1719432943863729?thread_ts=1719311116.215959&cid=C067EUS2D7S

**Review by mirnawong1 — APPROVED** [review]
great solution @mikealfare ! thanks so much and really appreciate this pr!  we don't have 1.3 in the docs dropdown since 1.3 is end of life so i wonder if we should just remove the content since it no longer applies for active/supported versions?

File: website/docs/reference/resource-configs/bigquery-configs.md [hunk 1]
```diff
@@ -706,9 +706,13 @@ models:
 
 Views with this configuration will be able to select from objects in `project_1.dataset_1` and `project_2.dataset_2`, even when they are located elsewhere and queried by users who do not otherwise have access to `project_1.dataset_1` and `project_2.dataset_2`.
 
+<VersionBlock lastVersion="1.3">
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/bigquery-configs.md]:
```suggestion
```

File: website/docs/reference/resource-configs/bigquery-configs.md [hunk 2]
```diff
@@ -706,9 +706,13 @@ models:
...
+
 #### Limitations
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/bigquery-configs.md]:
```suggestion
```

File: website/docs/reference/resource-configs/bigquery-configs.md [hunk 3]
```diff
@@ -706,9 +706,13 @@ models:
...
 
-The `grant_access_to` config _is thread-safe_. In versions prior to v1.4, it wasn't safe to use multiple threads for authorizing several views at once with `grant_access_to` for the same dataset. Initially, after adding a new `grant_access_to` setting, you can execute `dbt run` in a single thread. Later runs with the same configuration won't repeat the existing access grants and can use multiple threads.
+The `grant_access_to` config _is not thread-safe_. It isn't safe to use multiple threads for authorizing several views at once with `grant_access_to` for the same dataset. Initially, after adding a new `grant_access_to` setting, you should execute `dbt run` in a single thread. Later runs with the same configuration won't repeat the existing access grants and you can safely use multiple threads.
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/bigquery-configs.md]:
```suggestion
```

File: website/docs/reference/resource-configs/bigquery-configs.md [hunk 4]
```diff
@@ -706,9 +706,13 @@ models:
...
+
+</VersionBlock>
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/bigquery-configs.md]:
```suggestion
```

**Comment by mikealfare** [comment]
> great solution @mikealfare ! thanks so much and really appreciate this pr! we don't have 1.3 in the docs dropdown since 1.3 is end of life so i wonder if we should just remove the content since it no longer applies for active/supported versions?

Good catch. I've removed the Limitations section entirely since that was the only limitation.

**Review by mirnawong1 — APPROVED** [review]
thanks so much @mikealfare ! really appreciate you opening this pr and helping clarify the docs ✨

**Pull request merged**